components/search/src/filter.rs

```diff
@@ -221,7 +235,18 @@ fn find_engine_id(engines: &[SearchEngineDefinition], engine_id: String) -> Opti
     }
 }
 
-fn find_engine_with_match(
+fn set_engine_order(engines: &mut [SearchEngineDefinition], ordered_engines: &[String]) {
```

Comment:
I'm tempted to say we could put the sort related functions into a separate module, which might make it easier to write unit tests for them (and smaller module sizes), what do you think?